lib/repositories/folder.js

```diff
@@ -75,22 +75,30 @@ function createApi(table, provider, sqlGenerator, dependencies) {
     all: function() {
       dependencies.logger.trace('folder.all called');
 
-      var query = table
-        .select(table.star())
-        .from(table)
-        .toQuery();
-
-      return common.find(query, provider, constructor.bind(this))
+      return find(this)
```

Comment:
This may reuse some code, but it's less readable than it was before. A call to the find function passing only a this reference under the all method is confusing. I would revert all to what it was before and put the full where clause in the findByNameOrDTMF method. That way is method is self contained and uses a predictable API.